lib/rack/proxy.rb

```diff
@@ -39,7 +39,9 @@ def reconstruct_header_name(name)
     end
 
     # @option opts [String, URI::HTTP] :backend Backend host to proxy requests to
-    def initialize(opts = {})
+    def initialize(app = nil, opts= {})
+      @app = app
+      opts = app.is_a?(Hash) ? app.merge(opts) : opts
```

Comment:
Not sure why `app.merge(opts)`. In case the old way of calling things is used, this has no effect I think.

I would write it like this:
```
if app.is_a?(Hash)
  opts = app
  @app = nil
else
  @app = app
end
```

This way, we also avoid having incorrectly set `@app`.